docs/api/http-v1-batch.md

```diff
@@ -241,12 +184,102 @@ a sample request and response that includes one missing object:
 >     }
 <   ]
 < }
+```
+
+Object error codes should match HTTP status codes where possible:
+
+* 404 - The object does not exist on the server.
+* 410 - The object was removed by the owner.
+* 422 - Validation error.
 
-The error codes are integers that map to similar HTTP status codes where
-possible. Specifics can be included by the server in the `message` property.
+Validation errors can only occur on `upload` requests. Servers must verify
+that OIDs are valid SHA-256 strings, and that sizes are positive integers.
+Servers should also set an upper bound for the allowed object size too. Here's a
```

Comment:
I was purposely trying to use one of the rfc conformance words. Though after looking at the [RFC for the RFC conformance words](http://www.ietf.org/rfc/rfc2119.txt), I think you're right:

> SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.

> MAY   This word, or the adjective "OPTIONAL", mean that an item is
   truly optional.  One vendor may choose to include the item because a
   particular marketplace requires it or because the vendor feels that
   it enhances the product while another vendor may omit the same item.
   An implementation which does not include a particular option MUST be
   prepared to interoperate with another implementation which does
   include the option, though perhaps with reduced functionality. In the
   same vein an implementation which does include a particular option
   MUST be prepared to interoperate with another implementation which
   does not include the option (except, of course, for the feature the
   option provides.)